analyzers/src/SonarAnalyzer.Common/Rules/StringConcatenationInLoopBase.cs

```diff
@@ -100,50 +90,25 @@ private SyntaxNode GetInnerMostLeftOfConcatenation(TBinaryExpression binaryExpre
         private void CheckCompoundAssignment(SonarSyntaxNodeReportingContext context)
         {
             var addAssignment = (TAssignmentExpression)context.Node;
-            if (!IsString(GetLeft(addAssignment), context.SemanticModel))
-            {
-                return;
-            }
-
-            if (!TryGetNearestLoop(addAssignment, out var nearestLoop))
-            {
-                return;
-            }
 
-            if (context.SemanticModel.GetSymbolInfo(GetLeft(addAssignment)).Symbol is ILocalSymbol symbol &&
-                IsDefinedInLoop(GetLeft(addAssignment), nearestLoop, context.SemanticModel))
+            if (IsSystemString(Language.Syntax.AssignmentLeft(addAssignment), context.SemanticModel)
+                && TryGetNearestLoop(addAssignment, out var nearestLoop)
```

Comment:
`nearestLoop` is only needed in the conditional or branch. Inline it in that call to avoid unnecessary work or invert the or.